integration/query_comparison_compat_test.go

```diff
@@ -68,23 +68,55 @@ func TestQueryComparisonCompatImplicit(t *testing.T) {
 			filter:     bson.D{{"v.some.0", bson.A{42}}},
 			resultType: emptyResult,
 		},
+		"Int32": {
```

Comment:
Do we have a test for if _id is pushed down? I think we should have for each data type of ID.

Reply:
https://github.com/FerretDB/FerretDB/pull/1809/files#diff-eee22386d31e852a2d632d26269d2856e7c3fa7b58d03407c9ff6d344aba1b66R176 We do

Reply:
I think the filter you meant to add was `_id` instead of `v`. Something like this?
`filter: bson.D{{"_id", primitive.ObjectID{0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x10, 0x11}},`

`filter: bson.D{{"_id", primitive.ObjectID{0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01}},`

Also lets add one for int64, int32 and float64 :pray:

Reply:
Yeah, it seems so, although it's still worth to make a check for ObjectID for other keys than `_id`.

The checks for all the types for `_id` field are already implemented:

https://github.com/FerretDB/FerretDB/pull/1809/files#diff-b3f97e3a3514a7a7ff2a9104a0fd6e913c90b0eab5d079ea9e2ebcdeeb4e4b31R138-R164